core/web/resolver/query.go

```diff
@@ -371,3 +377,84 @@ func (r *Resolver) JobRun(ctx context.Context, args struct {
 
 	return NewJobRunPayload(&jr, r.App, err), nil
 }
+
+func (r *Resolver) ETHKeys(ctx context.Context) (*ETHKeysPayloadResolver, error) {
+	if err := authenticateUser(ctx); err != nil {
+		return nil, err
+	}
+
+	ks := r.App.GetKeyStore().Eth()
+
+	var keys []ethkey.KeyV2
+	var err error
+	if r.App.GetConfig().Dev() {
+		keys, err = ks.GetAll()
+	} else {
+		keys, err = ks.SendingKeys()
+	}
+	if err != nil {
+		return nil, fmt.Errorf("error getting unlocked keys: %v", err)
+	}
+
+	states, err := ks.GetStatesForKeys(keys)
+	if err != nil {
+		return nil, fmt.Errorf("error getting key states: %v", err)
+	}
+
+	var ethKeys []ETHKey
+
+	for _, state := range states {
+		k, err := ks.Get(state.Address.Hex())
+		if err != nil {
+			return nil, err
+		}
+
+		var gPrice *big.Int
+		var ethBalance *assets.Eth
+		var linkBalance *assets.Link
+
+		chain, err := r.App.GetChainSet().Get(state.EVMChainID.ToInt())
+		if errors.Cause(err) == evm.ErrNoChains {
+			ethKeys = append(ethKeys, ETHKey{
+				addr:      k.Address,
+				state:     state,
+				ethBal:    ethBalance,
+				linkBal:   linkBalance,
+				maxGPrice: gPrice,
+			})
+
+			continue
+		}
+		if err != nil {
+			return nil, fmt.Errorf("error getting EVM Chain: %v", err)
+		}
+
+		ethClient := chain.Client()
+		ethBal, err := ethClient.BalanceAt(ctx, state.Address.Address(), nil)
```

Comment:
Would be super nice to lean on the balance monitor here if we can

Reply:
This is great, I'll try and see if i can use it properly here!!